world-chain-builder/src/rpc/bundle.rs

```diff
@@ -0,0 +1,171 @@
+    Client: BlockReaderIdExt + StateProviderFactory + 'static,
+{
+    pub fn new(pool: Pool, client: Client) -> Self {
+        Self { pool, client }
+    }
+
+    pub fn provider(&self) -> &Client {
+        &self.client
+    }
+
+    pub fn pool(&self) -> &Pool {
+        &self.pool
+    }
+
+    /// Validates the conditional inclusion options provided by the client.
+    ///
+    /// reference for the implementation <https://notes.ethereum.org/@yoav/SkaX2lS9j#>
+    /// See also <https://pkg.go.dev/github.com/aK0nshin/go-ethereum/arbitrum_types#ConditionalOptions>
+    pub fn validate_options(&self, options: ConditionalOptions) -> RpcResult<()> {
+        let latest = self
+            .provider()
+            .block_by_id(BlockId::latest())
+            .map_err(|e| {
+                ErrorObject::owned(ErrorCode::InternalError.code(), e.to_string(), Some(""))
+            })?
+            .ok_or(ErrorObjectOwned::from(ErrorCode::InternalError))?;
+
+        self.validate_known_accounts(options.known_accounts, latest.header.number.into())?;
+
+        if let Some(min_block) = options.block_number_min {
+            if min_block > latest.number {
+                return Err(ErrorCode::from(-32003).into());
+            }
+        }
+
+        if let Some(max_block) = options.block_number_max {
+            if max_block <= latest.number {
+                return Err(ErrorCode::from(-32003).into());
+            }
+        }
+
+        if let Some(min_timestamp) = options.timestamp_min {
+            if min_timestamp > latest.timestamp {
+                return Err(ErrorCode::from(-32003).into());
+            }
+        }
+
+        if let Some(max_timestamp) = options.timestamp_max {
+            if max_timestamp <= latest.timestamp {
+                return Err(ErrorCode::from(-32003).into());
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Validates the account storage slots/storage root provided by the client
+    ///
+    /// Matches the current state of the account storage slots/storage root.
+    pub fn validate_known_accounts(
+        &self,
+        known_accounts: HashMap<Address, AccountStorage, FbBuildHasher<20>>,
+        latest: BlockId,
+    ) -> RpcResult<()> {
+        let state = self.provider().state_by_block_id(latest).map_err(|e| {
+            ErrorObject::owned(ErrorCode::InternalError.code(), e.to_string(), Some(""))
+        })?;
+
+        for (address, storage) in known_accounts.into_iter() {
+            match storage {
+                AccountStorage::Slots(slots) => {
+                    for (slot, value) in slots.into_iter() {
+                        let current = state.storage(address, slot.into()).map_err(|e| {
+                            ErrorObject::owned(
+                                ErrorCode::InternalError.code(),
+                                e.to_string(),
+                                Some(""),
```

Comment:
In cases where we are returning an empty string here, can we just return `None` instead?